engine/access/rpc/backend/backend_network.go

```diff
@@ -0,0 +1,121 @@
+package backend
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onflow/flow-go/access"
+
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+)
+
+var SnapshotHistoryLimitErr = fmt.Errorf("reached the snapshot history limit")
+
+type backendNetwork struct {
+	state                protocol.State
+	chainID              flow.ChainID
+	snapshotHistoryLimit int
+}
+
+// used by the observer
+func NewNetworkAPI(state protocol.State, chainID flow.ChainID, snapshotHistoryLimit int) *backendNetwork {
```

Comment:
Can you explain / add comments what is the purpose of this component and why we split it out?

Reply:
Added the comment. The observer needs to handle GetNetworkParameters, and GetLatestProtocolSnapshot the same way the access node does without having to setup the entire access backend.